### PR TITLE
refacto: Optimize loading

### PR DESCRIPTION
**Pull request by oktapodia**
Before:
image

After:
image

### CODE REVIEW

File: src/app/ui/app/App.tsx [hunk 1]
```diff
@@ -1,31 +1,48 @@
...
 
-const App = ({ starterVariant }: AppProps) => {
-  const { activeTheme, welcomeScreenClosed } = getCookies();
-  const isWelcomeScreenClosed = welcomeScreenClosed === 'true';
+const App = ({ starterVariant, isWelcomeScreenClosed, children }: AppProps) => {
+  const { trackEvent } = useUserTracking();
+
+  const welcomeScreen = useWelcomeScreen(isWelcomeScreenClosed);
+
+  const handleWelcomeScreenEnter = () => {
```
Comment (tcheee) [inline src/app/ui/app/App.tsx:22]: We have lost the hover effect on the WelcomeScreen, was it expected?

File: src/components/Widgets/Widgets.style.tsx [hunk 2]
```diff
@@ -3,137 +3,133 @@
...
+        : theme.spacing(3.5), // (mid viewheight - half-two/thirds widget height - navbar height )
     },
 
-    // widget wrappers -> animations
-    '& > .widget-wrapper > div:hover': !welcomeScreenClosed && {
-      marginTop: !welcomeScreenClosed ? 0 : theme.spacing(3.5),
+    [`@media screen and (min-height: 900px)`]: {
+      marginTop: !welcomeScreenClosed
+        ? 'calc( 50vh - 680px / 2.75 - 128px)'
+        : theme.spacing(3.5), // (mid viewheight - half-two/thirds widget height - ( navbar height + additional spacing) )
```
Comment (tcheee) [inline src/components/Widgets/Widgets.style.tsx:38]: might be these changes that affect the welcomeScreen hover?

File: src/components/Widgets/Widgets.tsx [hunk 3]
```diff
@@ -10,30 +9,19 @@ import type { WidgetSubvariant } from '@lifi/widget';
...
-export function Widgets({
-  widgetVariant,
-  activeTheme,
-  closedWelcomeScreen,
-}: WidgetsProps) {
+export function Widgets({ widgetVariant, closedWelcomeScreen }: WidgetsProps) {
   const { activeTab, setActiveTab } = useActiveTabStore();
-  const { welcomeScreenClosed, setWelcomeScreenClosed } =
-    useWelcomeScreen(closedWelcomeScreen);
+  const { setWelcomeScreenClosed } = useWelcomeScreen(closedWelcomeScreen);
```
Comment (tcheee) [inline src/components/Widgets/Widgets.tsx:23]: should not we rename this file?

File: src/components/Widgets/Widgets.tsx [hunk 4]
```diff
@@ -10,30 +9,19 @@ import type { WidgetSubvariant } from '@lifi/widget';
...
   const pathname = usePathname();
-  const { trackEvent } = useUserTracking();
   const [starterVariantUsed, setStarterVariantUsed] = useState(false);
   const [_starterVariant, setStarterVariant] = useState<
```
Comment (tcheee) [inline src/components/Widgets/Widgets.tsx:26]: do we still use this?

Comment (tcheee) [reply src/components/Widgets/Widgets.tsx:26]: might be another ticket to adapt this logic elsewhere for the tabs

File: src/app/ui/app/App.tsx (hunk 1 above)
Comment (oktapodia) [reply src/app/ui/app/App.tsx:24]: done

File: src/components/Widgets/Widgets.tsx (hunk 3 above)
Comment (oktapodia) [reply src/components/Widgets/Widgets.tsx:23]: No clue how to rename it :( any idea?

File: src/components/Widgets/Widgets.tsx (hunk 4 above)
Comment (oktapodia) [reply src/components/Widgets/Widgets.tsx:26]: A little bit for now, removing a part of it on each of my pr :s

File: src/app/[lng]/memecoins/page.tsx [hunk 5]
```diff
@@ -1,7 +1,26 @@
 import App from '@/app/ui/app/App';
+import { WidgetContainer, Widgets } from '@/components/Widgets';
+import { Widget } from '@/components/Widgets/Widget';
+import { getCookies } from '@/app/lib/getCookies';
 
 const Page = () => {
-  return <App starterVariant="default" />;
+  const variant = 'default';
```
Comment (dennyscode) [inline src/app/[lng]/memecoins/page.tsx:7]: We have this TabsMap.Exchange.variant. Wdyt?

File: src/app/[lng]/gas/page.tsx [hunk 6]
```diff
@@ -1,7 +1,25 @@
 import App from '@/app/ui/app/App';
+import { WidgetContainer, Widgets } from '@/components/Widgets';
+import { Widget } from '@/components/Widgets/Widget';
+import { getCookies } from '@/app/lib/getCookies';
 
-const Page = async () => {
-  return <App starterVariant="refuel" />;
+const Page = () => {
+  const variant = 'refuel';
```
Comment (dennyscode) [inline src/app/[lng]/gas/page.tsx:7]: We have this TabsMap.Refuel.variant. Wdyt?

File: src/app/[lng]/buy/page.tsx [hunk 7]
```diff
@@ -1,7 +1,26 @@
 import App from '@/app/ui/app/App';
+import { WidgetContainer, Widgets } from '@/components/Widgets';
+import { OnRamper } from '@/components/OnRamper';
+import { getCookies } from '@/app/lib/getCookies';
 
-const Page = async () => {
-  return <App starterVariant="buy" />;
+const Page = () => {
+  const variant = 'buy';
```
Comment (dennyscode) [inline src/app/[lng]/buy/page.tsx:7]: We have this TabsMap.Buy.variant. Wdyt?

File: src/app/[lng]/page.tsx [hunk 8]
```diff
@@ -1,5 +1,24 @@
 import App from '../ui/app/App';
+import { WidgetContainer, Widgets } from '@/components/Widgets';
+import { Widget } from '@/components/Widgets/Widget';
+import { getCookies } from '@/app/lib/getCookies';
+import { WidgetSkeleton } from '@/components/Widgets/WidgetSkeleton';
 
 export default function Page() {
-  return <App starterVariant="default" />;
+  const variant = 'default'; // exchange
```
Comment (dennyscode) [inline src/app/[lng]/page.tsx:8]: We have this TabsMap.Exchange.variant. Wdyt?

File: src/components/WelcomeScreen/WelcomeScreen.style.ts [hunk 9]
```diff
@@ -36,19 +41,20 @@ const fadeOut = keyframes`
...
-})<ContentWrapperProps>(({ theme, showWelcome }) => ({
+export const ContentWrapper = styled(
+  Box,
+  {},
+)<ContentWrapperProps>(({ theme }) => ({
   textAlign: 'center',
   background: theme.palette.mode === 'dark' ? '#1A1033' : '#F3EBFF',
   width: '100%',
-  position: 'absolute',
+  // position: 'absolute',
```
Comment (dennyscode) [inline src/components/WelcomeScreen/WelcomeScreen.style.ts:51]: Let´s remove those comments if not needed

File: src/components/WelcomeScreen/WelcomeScreen.style.ts [hunk 10]
```diff
@@ -36,19 +41,20 @@ const fadeOut = keyframes`
...
   // top: 0,
-  bottom: 0,
+  // bottom: 0,
   zIndex: '1400',
   height: 'auto',
-  animation: !showWelcome ? fadeOut : 'unset',
-  animationDuration: '.5s',
+  // animation: !showWelcome ? fadeOut : 'unset',
+  // animationDuration: '.5s',
```
Comment (dennyscode) [inline src/components/WelcomeScreen/WelcomeScreen.style.ts:57]: Let´s remove those comments if not needed

File: src/components/Widgets/Widget.tsx [hunk 11]
```diff
@@ -53,6 +55,15 @@ export function Widget({
     enabled: !!themeVariant,
   });
 
+  const router = useRouter();
+
+  useEffect(() => {
+    console.log('prefetching');
```
Comment (dennyscode) [inline src/components/Widgets/Widget.tsx:61]: Do we need this log?

File: src/components/Widgets/WidgetSkeleton.tsx [hunk 12]
```diff
@@ -4,17 +4,15 @@ import { Box } from '@mui/material';
...
-  welcomeScreenClosed,
-  config,
-}: WidgetSkeletonProps) => {
+export const WidgetSkeleton = ({ config }: WidgetSkeletonProps) => {
   return (
     <Box height="682px">
-      <WidgetSkeletonContainer welcomeScreenClosed={welcomeScreenClosed}>
+      <WidgetSkeletonContainer welcomeScreenClosed={true}>
+        {' '}
+        {/* Does not matter in this scenario */}
```
Comment (dennyscode) [inline src/components/Widgets/WidgetSkeleton.tsx:15]: Is this needed?

**Comment by dennyscode** [comment]
One more problem here: Widget does not seem to show full width when expanded.
image

**Comment by dennyscode** [comment]
Also, unfortunately we do not maintain the state of a tab now:

[maintain-widget-states-screen-capture.webm](https://github.com/lifinance/jumper.exchange/assets/43956540/7243601c-922b-436c-bad7-4ec2e4ba1a44)

File: src/components/Menu/SubMenu.style.ts [hunk 13]
```diff
@@ -13,7 +13,7 @@ export interface SubMenuLabelProps extends Omit<TypographyProps, 'component'> {
 
 export const SubMenuLabel = styled(Typography, {
   shouldForwardProp: (prop) =>
-    prop !== 'prefixIcon' && prop !== 'suffixIcon' && prop !== 'component',
+    prop !== 'isPrefixIcon' && prop !== 'isSuffixIcon' && prop !== 'component',
```
Comment (dennyscode) [inline src/components/Menu/SubMenu.style.ts:16]: Nice! I also fixed that in another ticket, good find!

File: src/components/Navbar/NavbarTabs/NavbarTabs.tsx [hunk 14]
```diff
@@ -45,7 +45,7 @@ export const NavbarTabs = ({ navbarPageReload }: NavbarTabsProps) => {
   return (
     <Tabs
       data={navbarTabs}
-      value={!isDesktop || navbarPageReload ? -1 : activeTab}
```
Comment (dennyscode) [inline src/components/Navbar/NavbarTabs/NavbarTabs.tsx:48]: image

This change causes to show an active tab on the profile-page.

Comment (dennyscode) [reply src/components/Navbar/NavbarTabs/NavbarTabs.tsx:48]: Btw: In this ticket, I have touched the wrong value of -1 for the tabs. I found that `false` can be passed to deactivate the active tab. For that, the types needed some cleanup:

https://github.com/lifinance/jumper.exchange/pull/1029

**Review by dennyscode — CHANGES_REQUESTED** [review]

File: src/app/[lng]/memecoins/page.tsx (hunk 5 above)
Comment (dennyscode) [reply src/app/[lng]/memecoins/page.tsx:7]: I see, alright then!

File: src/app/[lng]/gas/page.tsx (hunk 6 above)
Comment (oktapodia) [reply src/app/[lng]/gas/page.tsx:7]: https://github.com/lifinance/jumper.exchange/pull/1023#discussion_r1639450829

File: src/app/[lng]/buy/page.tsx (hunk 7 above)
Comment (oktapodia) [reply src/app/[lng]/buy/page.tsx:7]: https://github.com/lifinance/jumper.exchange/pull/1023#discussion_r1639450829

File: src/app/[lng]/page.tsx (hunk 8 above)
Comment (oktapodia) [reply src/app/[lng]/page.tsx:8]: https://github.com/lifinance/jumper.exchange/pull/1023#discussion_r1639450829

**Review by dennyscode — APPROVED** [review]

**Review by tcheee — APPROVED** [review]

**Pull request merged**